test/support/utils_sort.h

```diff
@@ -0,0 +1,473 @@
+}
+
+template <typename Policy, typename InputIterator, typename OutputIterator, typename OutputIterator2, typename Size,
+          typename... Compare>
+oneapi::dpl::__internal::__enable_if_hetero_execution_policy<Policy>
+run_test(SortTestConfig config,
+         Policy&& exec, OutputIterator tmp_first, OutputIterator tmp_last, OutputIterator2 expected_first,
+         OutputIterator2 expected_last, InputIterator first, InputIterator last, Size n, Compare ...compare)
+{
+    // Run tests for USM shared memory (external testing for USM shared memory, once already covered in sycl_iterator.pass.cpp)
+    if (config.test_usm_shared)
+    {
+        test_usm<sycl::usm::alloc::shared>(config, std::forward<Policy>(exec), tmp_first, tmp_last,
+                                           expected_first, expected_last,first, last, n, compare...);
+    }
+    if (config.test_usm_device)
+    {
+        // Run tests for USM device memory
+        test_usm<sycl::usm::alloc::device>(config, std::forward<Policy>(exec), tmp_first, tmp_last,
+                                           expected_first, expected_last, first, last, n, compare...);
+    }
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT && _PSTL_SYCL_TEST_USM
+
+template <typename T>
+struct test_sort_op
+{
+    SortTestConfig config;
+
+    template <typename Policy, typename InputIterator, typename OutputIterator, typename OutputIterator2, typename Size,
+              typename... Compare>
+    std::enable_if_t<
+        TestUtils::is_base_of_iterator_category_v<std::random_access_iterator_tag, InputIterator> &&
+            (TestUtils::can_use_default_less_operator_v<T> || sizeof...(Compare) > 0)>
+    operator()(Policy&& exec, OutputIterator tmp_first, OutputIterator tmp_last, OutputIterator2 expected_first,
+               OutputIterator2 expected_last, InputIterator first, InputIterator last, Size n, Compare ...compare)
+    {
+        run_test(config, std::forward<Policy>(exec), tmp_first, tmp_last, expected_first, expected_last,
+                first, last, n, compare...);
+    }
+
+    template <typename Policy, typename InputIterator, typename OutputIterator, typename OutputIterator2, typename Size,
+              typename... Compare>
+    std::enable_if_t<
+        !TestUtils::is_base_of_iterator_category_v<std::random_access_iterator_tag, InputIterator> ||
+            !(TestUtils::can_use_default_less_operator_v<T> || sizeof...(Compare) > 0)>
+    operator()(Policy&& /* exec */, OutputIterator /* tmp_first */, OutputIterator /* tmp_last */,
+               OutputIterator2 /* expected_first */, OutputIterator2 /* expected_last */, InputIterator /* first */,
+               InputIterator /* last */, Size /* n */, Compare .../*compare*/)
+    {
+    }
+};
+
+template <typename T, typename Invoker, typename Converter, typename... Compare>
+void
+test_sort(SortTestConfig config, const std::vector<std::size_t> sizes, Invoker invoker, Converter converter,
+          Compare... compare)
```

Comment:
```suggestion
test_sort(SortTestConfig config, const std::vector<std::size_t>& sizes, Invoker invoker, Converter converter,
          Compare... compare)
```
- please avoid to pass `std::vector` by value

Reply:
Done